Allow variant autocomplete result filtering by stock availability

richardnuno opened the pull request:
When creating a stock transfer the variant autocomplete results should filter variants that do not have any stock.

athal7 commented on api/app/controllers/spree/api/variants_controller.rb:
```diff
@@ -66,6 +66,7 @@ def scope
               variants = variants.active
             end
           end
+          variants = variants.in_stock if params[:hide_out_of_stock]
```
:+1:

athal7 commented:
i'm finding it a bit challenging to wrap my head around the double negatives here. what do you think about having the param be 'in_stock_only' ?

richardnuno commented:
sure, sounds good

richardnuno commented:
@athal7 updated

gmacdougall commented:
:+1:

athal7 commented:
:+1:

richardnuno commented:
@athal7 @gmacdougall one small change in the last commit, added the `in_stock_only: true` option to the variant autocomplete used when creating orders in the admin. This way, only variants that can successfully be purchased are displayed. Mind taking another look?

athal7 commented:
:+1:

richardnuno commented:
@athal7 fixed tests in latest commit

athal7 commented:
:+1:

gmacdougall commented:
:+1:

richardnuno commented:
Closing in favor of https://github.com/bonobos/spree/pull/377 since this PR was opened against the create-stock-transfer branch. PR contains the exact same changes, just squashed and rebased on top of 2-2-dev.
